src/deepEqual.js

```diff
@@ -20,7 +20,21 @@
  * deepEqual({test: {abc: 5}}, {test: {abc: 5, def: 4}}) === false
  */
 function deepEqual(a, b) {
-  // write code here
-}
-
+  if (typeof a === 'object' && typeof b === 'object' && a !== null && b !== null) {
+    if (Object.keys(a).length !== Object.keys(b).length) {
+      return false;
+    }
+    for (let key in a) {
+      if (!b.hasOwnProperty(key)) {
+        return false;
+      } else if (b.hasOwnProperty(key) && b[key] === a[key]) {
+        continue;
+      } else if (b.hasOwnProperty(key) && b[key] !== a[key]) {
+        return deepEqual(b[key], a[key]);
```

Comment:
if values for the first key in 'a' are equal, then we'll quit the loop and won't compare values for other keys.